### ISSUE TITLE
XAML Hot Reload Support

### ISSUE THREAD

**Issue by Bush-cat** [w-ahmad/WinUI.TableView#20]
Could we possibly fix XAML hot reload?

When you make changes to the Xaml while using Hot Reload, you will encounter a Warning:
`This change is not supported by XAML Hot Reload and will not be applied during the debugging session.`
or Error: `Failed to retrieve the child at index 5.`

Also the debugger stops working when using the live preview on the tables.

Thank you for your work on TableView!
This Addition is not essential for me but it would make working in XAML easier.

**Comment by w-ahmad** [comment]
I believe this is the limitation of the HotReload for the WinUI at the time. I tried a couple of things, but this doesn't work.

**Comment by w-ahmad** [comment]
@michael-hawker Is there a workaround to resolve this issue?

**Comment by michael-hawker** [comment]
Any collections of XAML things in XAML should inherit from `DependencyObjectCollection` (see things like the animation sets and SwitchPresenter cases in WCT). So, as you mentioned previously, that should be the solution. (It's similar to the reasons you should use DependencyProperty within XAML objects vs. INotifyPropertyChanged, as per my video the other month. The former is within the XAML framework, the later is for applications and data using XAML.)

**Comment by w-ahmad** [comment]
Thanks a lot for the clear explanation and references, @michael-hawker!
This really helps clarify why DependencyObjectCollection is needed for XAML support and Hot Reload.
My only concern is that using DependencyObjectCollection means we lose type safety for the collection.

<img width="822" height="577" alt="Image" src="https://github.com/user-attachments/assets/ed7012a3-2f6a-48cc-9d18-133711d76439" />

**Comment by michael-hawker** [comment]
Yes, that is indeed a limitation/drawback of `DependencyObjectCollection`, I haven't tried messing around lately with if there's some other workarounds. `IList` is a special type in the XAML world and handled differently sometimes too. So, using that as a base or interface could change things too:

```cs
public partial class TableViewColumnsCollection : DependencyObjectCollection, IList<TableViewColumn>
```

I did a quick initial quick test and was able to get things compiled for SwitchPresenter this way, but was hard to see how VS intellisensee/hotreload reacted, as I think there's some other issue with our WASDK sample there.

**Comment by w-ahmad** [comment]
I tried a little different here to ensure type safety in XAML and code. Here is my implementation:
```cs
// Interface with some other properties like 'TableView', 'VisibleColumns'
public interface ITableViewColumnsCollection : IList<TableViewColumn> {  }

// Implementation
public partial class TableViewColumnsCollection : DependencyObjectCollection, ITableViewColumnsCollection {  }

// Actual Columns property in TableView
public ITableViewColumnsCollection Columns { get; } = new TableViewColumnsCollection();
```
With this, XAML Hot Reload is working well and provides accurate IntelliSense.

Thank you so much, @michael-hawker! It would have been much harder to make it happen without your guidance.
